crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -133,6 +152,62 @@ impl Epoch {
         ))
     }
 
+    /// Information about whether last epoch change used safe mode, which happens if the full epoch
+    /// change logic fails for some reason.
+    async fn safe_mode(&self) -> Option<SafeMode> {
+        Some(SafeMode {
+            enabled: Some(self.system_state.safe_mode),
+            gas_summary: Some(GasCostSummary {
+                computation_cost: self.system_state.safe_mode_computation_rewards,
+                storage_cost: self.system_state.safe_mode_storage_rewards,
+                storage_rebate: self.system_state.safe_mode_storage_rebates,
+                non_refundable_storage_fee: self.system_state.safe_mode_non_refundable_storage_fee,
+            }),
+        })
+    }
+
+    /// The value of the `version` field of `0x5`, the `0x3::sui::SuiSystemState` object.  This
+    /// version changes whenever the fields contained in the system state object (held in a dynamic
+    /// field attached to `0x5`) change.
+    async fn system_state_version(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.system_state.system_state_version))
+    }
+
+    /// Details of the system that are decided during genesis.
+    async fn system_parameters(&self) -> Option<SystemParameters> {
+        Some(SystemParameters {
+            duration_ms: Some(BigInt::from(self.system_state.epoch_duration_ms)),
+            stake_subsidy_start_epoch: Some(self.system_state.stake_subsidy_start_epoch),
+            min_validator_count: Some(self.system_state.max_validator_count),
```

Comment:
I vaguely remember a discussion about this from before where it was hard to extract the true `min_validator_count` is that right, or is this a typo?  If the former, let's add a comment explaining the context, to avoid future heart attacks.

Reply:
Good catch!